### PR TITLE
fix: treat textchange as single character chunk , use get_cursor instead of marks [ and ]

### PR DESCRIPTION
**Pull request by ArbitRandomUser**

### CODE REVIEW

File: lua/smuggler/buffers.lua [hunk 1]
```diff
@@ -223,12 +223,19 @@ end
 --- this triggers the invalidation of the block even though it shouldn't...
 function M.invalidate_changed_chunks(buffer)
 	log.debug("Invalidating!")
-	local tmp = vim.api.nvim_buf_get_mark(buffer.number, "[")
-	local rowstart = tmp[1]
-	local colstart = tmp[2]
-	tmp = vim.api.nvim_buf_get_mark(buffer.number, "]")
-	local rowstop = tmp[1]
-	local colstop = tmp[2]
+	--local tmp = vim.api.nvim_buf_get_mark(buffer.number, "[")
```
Comment (Klafyvel) [inline lua/smuggler/buffers.lua:226]: Could you remove these commented code?

File: lua/smuggler/buffers.lua [hunk 2]
```diff
@@ -223,12 +223,19 @@ end
...
+	--local colstart = tmp[2]
+	--tmp = vim.api.nvim_buf_get_mark(buffer.number, "]")
+	--local rowstop = tmp[1]
+	--local colstop = tmp[2]
+
+    local cur = vim.api.nvim_win_get_cursor(0)
+    local rowstart = cur[1]
+    local colstart = cur[2]
+	local rowstop = cur[1]
+	local colstop = cur[2]
```
Comment (Klafyvel) [inline lua/smuggler/buffers.lua:237]: Could you fix the indentation here? (I'll add a code formatter pass in the CI at some point when I write tests for the plugin)

**Review by Klafyvel — CHANGES_REQUESTED** [review]
Hi! Thank you for this. I must say that my first reaction when I read the code was "This will never work in normal mode!" but you proved me wrong, and that's amazing! I have only tiny cosmetic comments to make. Can you add a `Release-As: 0.4.2` in your next commit to make it a patch release ? See [here](https://github.com/googleapis/release-please?tab=readme-ov-file#how-do-i-change-the-version-number).

**Comment by ArbitRandomUser** [comment]
i'll get to it soon , 
i was thinking we treat every character change as a chunk, find intersecting chunks in buffer and invalidate them. Since a single character change will occur only inside one chunk of the buffer, finding all intersecting chunks is unnecessary. Maybe we refactor this to just find the current chunk the cursor is in ?

**Comment by Klafyvel** [comment]
Ok, my mistake I found a problem with this approach, and indeed it shows in normal mode. If your changes overlap the evaluated chunk, it is not detected. Case in point: Visual-block select a portion of code that contains an evaluated chunk and add a space at the begining of each line. Because the cursor is never in the chunk we miss the modification. Perhaps we should have two detection mechanisms: one cursor-based for insertion mode, and one mark-based for other modes.

**Comment by ArbitRandomUser** [comment]
sorry for all the indentation mixups , just realized the file is tab indented and ive setup vim to expand tabs

**Comment by ArbitRandomUser** [comment]
actually it seems to have a mix of both, 
fig

**Comment by Klafyvel** [comment]
Thanks, I'll have a look tonight!

**Review by Klafyvel — APPROVED** [review]

**Comment by Klafyvel** [comment]
Thank you very much for your help on this!

**Pull request merged**